File: fix_sticky_buttons.py
import re

with open('/Users/aliceer/UDO/frontend/src/main.js', 'r') as f:
    content = f.read()

start_marker = '  <!-- Sticky Right Buttons (Why UDO & LINE) -->'
end_marker = '  </div>\n'

start_idx = content.find(start_marker)
if start_idx != -1:
    end_idx = content.find(end_marker, start_idx) + len(end_marker)
    
    new_buttons = """  <!-- Sticky Right Buttons (Why UDO & LINE) -->
  <div class="fixed bottom-6 md:bottom-8 right-6 md:right-8 z-[100] flex flex-col items-center gap-3">
    
    <!-- Why UDO Button -->
    <div class="relative group cursor-pointer" id="whyUdoBtnContainer">
      <!-- Close Button (X) -->
      <button onclick="document.getElementById('whyUdoBtnContainer').style.display='none'" class="absolute -top-1 -right-1 w-6 h-6 md:w-7 md:h-7 bg-black/70 text-white rounded-full flex items-center justify-center hover:bg-black/90 transition-colors z-10 border-[1.5px] border-white/30 shadow-sm">
        <svg xmlns="http://www.w3.org/2000/svg" class="w-3.5 h-3.5 md:w-4 md:h-4" fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2.5">
          <path stroke-linecap="round" stroke-linejoin="round" d="M6 18L18 6M6 6l12 12" />
        </svg>
      </button>
      
      <!-- Main Circle Button -->
      <!-- ขยายปุ่มให้ใหญ่ขึ้น (w-100px) และเปลี่ยนตัวหนังสือให้บางลง (font-medium) -->
      <a href="#" class="w-[84px] h-[84px] md:w-[100px] md:h-[100px] bg-[#E32626] hover:bg-[#c92020] rounded-full flex flex-col items-center justify-center shadow-[0_10px_20px_rgba(227,38,38,0.2)] transition-transform hover:scale-105 duration-300">
        <span class="text-white text-[13px] md:text-[15px] leading-[1.3] text-center font-medium">ทำไมต้อง<br>ซื้อกับ<br>UDO</span>
      </a>
    </div>

    <!-- LINE Button -->
    <!-- เพิ่มเส้นขอบสีขาว (border-2 border-white) และขยายปุ่มนิดนึง -->
    <a href="#" class="w-[52px] h-[52px] md:w-[60px] md:h-[60px] bg-[#E32626] hover:bg-[#00B900] rounded-full flex items-center justify-center shadow-lg transition-all hover:scale-110 duration-300 relative group border-[2px] border-white">
      <!-- LINE SVG Icon (ขยายขนาดจาก w-8 เป็น w-10 เพื่อให้คำว่า LINE ข้างในอ่านออกชัดเจนขึ้น) -->
      <svg class="w-8 h-8 md:w-10 md:h-10 text-white relative z-10" viewBox="0 0 24 24" fill="currentColor">
        <path d="M24 10.304c0-5.369-5.383-9.738-12-9.738-6.616 0-12 4.369-12 9.738 0 4.814 4.269 8.846 10.036 9.608.391.084.922.258 1.057.592.121.303.079.778.039 1.085l-.171 1.027c-.053.303-.242 1.186 1.039.647 1.281-.54 6.911-4.069 9.428-6.967 1.739-1.907 2.572-3.843 2.572-5.992zM8.344 13.56H6.173c-.29 0-.524-.234-.524-.524v-4.48c0-.289.234-.523.524-.523h2.171c.29 0 .524.234.524.523v1.308c0 .29-.234.524-.524.524H6.697v1.865h1.647c.29 0 .524.234.524.523v1.308c0 .29-.234.524-.524.524H8.344zm3.924 0h-1.646c-.29 0-.524-.234-.524-.524v-4.48c0-.289.234-.523.524-.523h1.646c.29 0 .524.234.524.523v4.48c0 .29-.234.524-.524.524zm4.184 0h-1.288l-1.921-2.617v2.093c0 .29-.234.524-.524.524h-1.647c-.289 0-.523-.234-.523-.524v-4.48c0-.289.234-.523.523-.523h1.289l1.92 2.617V8.553c0-.289.234-.523.524-.523h1.647c.289 0 .523.234.523.523v4.48c0 .29-.234.524-.523.524zm4.07 0h-2.17c-.29 0-.524-.234-.524-.524v-4.48c0-.289.234-.523.524-.523h2.17c.289 0 .523.234.523.523v1.308c0 .29-.234.524-.523.524h-1.646v1.08h1.646c.289 0 .523.234.523.524v1.308c0 .29-.234.524-.523.524h-1.646v1.308c0 .29.234.524.523.524h2.17c.289 0 .524.234.524.524z" />
      </svg>
    </a>
    
  </div>\n"""
    
    content = content[:start_idx] + new_buttons + content[end_idx:]
    with open('/Users/aliceer/UDO/frontend/src/main.js', 'w') as f:
        f.write(content)
    print("Fixed buttons successfully.")
else:
    print("Could not find the start marker.")

File: fix_sticky_buttons.js
const fs = require('fs');
let content = fs.readFileSync('frontend/src/main.js', 'utf8');

// 1. Remove the two buttons I just added
const wrongButtonsRegex = /<!-- Floating "Why buy with UDO" Button \(Bottom Left\) -->[\s\S]*?<!-- Floating LINE Button \(Bottom Right\) -->[\s\S]*?<\/a>\n\n/g;
content = content.replace(wrongButtonsRegex, '');

// 2. Insert the LINE button into the existing Sticky Right Buttons container
const containerRegex = /<!-- Main Circle Button \(ย่อลงมานิดนึง เป็น 108px\) -->\n\s*<a href="#" class="w-\[90px\] h-\[90px\] md:w-\[108px\] md:h-\[108px\] bg-\[#E12427\] hover:bg-\[#c41f22\] rounded-full flex flex-col items-center justify-center shadow-\[0_10px_20px_rgba\(181,56,50,0\.25\)\] transition-transform hover:scale-105 duration-300">\n\s*<span class="text-white text-\[14px\] md:text-\[17px\] leading-\[1\.3\] text-center font-medium">ทำไมต้อง<br>ซื้อกับ<br>UDO<\/span>\n\s*<\/a>\n\s*<\/div>\n\s*<\/div>/;

const newContainer = `<!-- Main Circle Button (ย่อลงมานิดนึง เป็น 108px) -->
      <a href="#" class="w-[90px] h-[90px] md:w-[108px] md:h-[108px] bg-[#E12427] hover:bg-[#c41f22] rounded-full flex flex-col items-center justify-center shadow-[0_10px_20px_rgba(181,56,50,0.25)] transition-transform hover:scale-105 duration-300">
        <span class="text-white text-[14px] md:text-[17px] leading-[1.3] text-center font-medium">ทำไมต้อง<br>ซื้อกับ<br>UDO</span>
      </a>
    </div>
    
    <!-- Floating LINE Button (Bottom Right) -->
    <a href="#" class="w-[54px] h-[54px] md:w-[60px] md:h-[60px] bg-[#E12427] rounded-full border-[3px] border-white shadow-[0_8px_24px_rgba(225,36,39,0.4)] flex items-center justify-center hover:scale-110 hover:shadow-[0_12px_32px_rgba(225,36,39,0.5)] transition-all duration-300 cursor-pointer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512" class="w-8 h-8 md:w-9 md:h-9 text-white" fill="currentColor">
        <path d="M448 240.2c0-98.8-100.5-179.2-224-179.2S0 141.4 0 240.2c0 88.5 78.4 162.7 181.8 176.4 11.2 2.3 26 7.2 29.8 16.6 2.9 7 1.4 18 1 24.3-.4 6-2 19.3 16.9 11.3 18.9-7.9 102.1-60 140.9-104 30-34.1 77.6-80.1 77.6-124.6zm-175.9 35.1v71.1c0 1.8-1.4 3.2-3.2 3.2h-11.4c-1.1 0-2.1-.6-2.6-1.3l-32.6-44v42.2c0 1.8-1.4 3.2-3.2 3.2h-11.4c-1.8 0-3.2-1.4-3.2-3.2v-71.1c0-1.8 1.4-3.2 3.2-3.2H219c1 0 2.1.5 2.6 1.4l32.6 44v-42.2c0-1.8 1.4-3.2 3.2-3.2h11.4c1.8-.1 3.3 1.4 3.3 3.1zm-82-3.2h-11.4c-1.8 0-3.2 1.4-3.2 3.2v71.1c0 1.8 1.4 3.2 3.2 3.2h11.4c1.8 0 3.2-1.4 3.2-3.2v-71.1c0-1.7-1.4-3.2-3.2-3.2zm-27.5 59.6h-31.1v-56.4c0-1.8-1.4-3.2-3.2-3.2h-11.4c-1.8 0-3.2 1.4-3.2 3.2v71.1c0 .9.3 1.6.9 2.2.6.5 1.3.9 2.2.9h45.7c1.8 0 3.2-1.4 3.2-3.2v-11.4c0-1.7-1.4-3.2-3.1-3.2zM332.1 201h-45.7c-1.7 0-3.2 1.4-3.2 3.2v71.1c0 1.7 1.4 3.2 3.2 3.2h11.4c1.8 0 3.2-1.4 3.2-3.2v-20h31.1c1.8 0 3.2-1.4 3.2-3.2v-11.4c0-1.8-1.4-3.2-3.2-3.2h-31.1v-14h31.1c1.8 0 3.2-1.4 3.2-3.2v-11.4c-.1-1.7-1.5-3.1-3.2-3.1z"/>
      </svg>
    </a>
  </div>`;

if (content.match(containerRegex)) {
    content = content.replace(containerRegex, newContainer);
    fs.writeFileSync('frontend/src/main.js', content, 'utf8');
    console.log("Updated sticky buttons successfully.");
} else {
    console.log("Could not find container regex.");
}
